File: src/main/java/main/users/all/Administrators.java
package main.users.all;

import main.tenderInformation.TenderDescription;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

public class Administrators {
    private WebDriver driver;

    public Administrators(WebDriver driver) {
        this.driver = driver;
    }

    private By addAdministratorLink = By.xpath("//span[@class=\"contentHeadingLink1\" and contains(text(), \"Tender administrator\") and contains(@onclick, \"document.data.usertype.value='4'\")]");
    private By addObserverLink = By.xpath("//span[@class=\"contentHeadingLink1\" and normalize-space()=\"Tender observer\" and @onclick=\"document.data.usertype.value='8';saction();\"]");
    private By addDocumentCoordinatorLink = By.xpath("//span[contains(text(), 'Tender observer')][@class=\"contentHeadingLink1\"][@onclick=\"document.data.usertype.value='8';saction();\"]");
    private By previouslyInvitedLink = By.xpath("//a[contains(text(), 'Select previously invited users')][@class=\"contentHeading1\"][@onclick=\"select_invited();\"]");
    private By textFieldForEmail = By.xpath("//textarea[@name='emails']");
    private By userToUpdateProfile = By.xpath("//td[text()='yaryna bilan']");
    private By adminProfileDropdown = By.xpath("//select[@name='adminprofile']");
    private By tenderAdminProfileDropdown = By.xpath("//option[text()='Tender administrator']");
    private By accessTypeDropdown = By.xpath("//option[text()='Full access']\n");
    private By okButton = By.xpath("//input[@name='button_ok']\n");
    public void clickAddAdministratorLink() {
        WebElement addAdministratorLinkElement = driver.findElement(addAdministratorLink);
        addAdministratorLinkElement.click();
    }

    public void clickAddObserverLink() {
        WebElement addObserverLinkElement = driver.findElement(addObserverLink);
        addObserverLinkElement.click();
    }

    public void clickAddDocumentCoordinatorLink() {
        WebElement addDocumentCoordinatorLinkElement = driver.findElement(addDocumentCoordinatorLink);
        addDocumentCoordinatorLinkElement.click();
    }
    public void clickOnUserToUpdateProfile() {
        WebElement userToUpdateProfileElement = driver.findElement(userToUpdateProfile);
        userToUpdateProfileElement.click();
    }

    public void clickPreviouslyInvitedLink() {
        WebElement previouslyInvitedLinkElement = driver.findElement(previouslyInvitedLink);
        previouslyInvitedLinkElement.click();
    }

    public Administrators fillInTextFieldForEmailInput(String textFieldForEmail) {
        WebElement textFieldForEmailInput = textFieldForEmailInput();
        textFieldForEmailInput.sendKeys(textFieldForEmail);
        return this;
    }
    private WebElement textFieldForEmailInput() {
        return driver.findElement(textFieldForEmail);
    }

    public void clickOnAdminProfileDropdown() {
        WebElement dropdown = driver.findElement(adminProfileDropdown);
        dropdown.click();
    }

    public void selectTenderAdminProfile() {
        WebElement dropdown = driver.findElement(tenderAdminProfileDropdown);
        dropdown.click();
    }

    public void selectFullAccessType() {
        WebElement dropdown = driver.findElement(accessTypeDropdown);
        dropdown.click();
    }


}

File: src/main/java/main/users/all/Administrators.kt
package main.users.all

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import org.openqa.selenium.WebElement

class Administrators(private val driver: WebDriver) {
    private val addAdministratorLink: By = By.xpath("//span[@class=\"contentHeadingLink1\" and contains(text(), \"Tender administrator\") and contains(@onclick, \"document.data.usertype.value='4'\")]")
    private val addObserverLink: By = By.xpath("//span[@class=\"contentHeadingLink1\" and normalize-space()=\"Tender observer\" and @onclick=\"document.data.usertype.value='8';saction();\"]")
    private val addDocumentCoordinatorLink: By = By.xpath("//span[contains(text(), 'Tender observer')][@class=\"contentHeadingLink1\"][@onclick=\"document.data.usertype.value='8';saction();\"]")
    private val previouslyInvitedLink: By = By.xpath("//a[contains(text(), 'Select previously invited users')][@class=\"contentHeading1\"][@onclick=\"select_invited();\"]")
    private val textFieldForEmail: By = By.xpath("//textarea[@name='emails']")
    private val userToUpdateProfile: By = By.xpath("//td[text()='yaryna bilan']")
    private val adminProfileDropdown: By = By.xpath("//select[@name='adminprofile']")
    private val tenderAdminProfileDropdown: By = By.xpath("//option[text()='Tender administrator']")
    private val accessTypeDropdown: By = By.xpath("//option[text()='Full access']\n")
    private val okButton: By = By.xpath("//input[@name='button_ok']\n")

    fun clickAddAdministratorLink() {
        val addAdministratorLinkElement = driver.findElement(addAdministratorLink)
        addAdministratorLinkElement.click() }

    fun clickAddObserverLink() {
        val addObserverLinkElement = driver.findElement(addObserverLink)
        addObserverLinkElement.click() }

    fun clickAddDocumentCoordinatorLink() {
        val addDocumentCoordinatorLinkElement = driver.findElement(addDocumentCoordinatorLink)
        addDocumentCoordinatorLinkElement.click() }

    fun clickOnUserToUpdateProfile() {
        val userToUpdateProfileElement = driver.findElement(userToUpdateProfile)
        userToUpdateProfileElement.click() }

    fun clickPreviouslyInvitedLink() {
        val previouslyInvitedLinkElement = driver.findElement(previouslyInvitedLink)
        previouslyInvitedLinkElement.click() }

    fun fillInTextFieldForEmailInput(textFieldForEmail: String?): Administrators {
        val textFieldForEmailInput = textFieldForEmailInput()
        textFieldForEmailInput.sendKeys(textFieldForEmail)
        return this }

    private fun textFieldForEmailInput(): WebElement {
        return driver.findElement(textFieldForEmail) }

    fun clickOnAdminProfileDropdown() {
        val dropdown = driver.findElement(adminProfileDropdown)
        dropdown.click() }

    fun selectTenderAdminProfile() {
        val dropdown = driver.findElement(tenderAdminProfileDropdown)
        dropdown.click() }

    fun selectFullAccessType() {
        val dropdown = driver.findElement(accessTypeDropdown)
        dropdown.click()}
}